File: ga/4omini_result_2024_7/result/1968C_cpp_mini/16_1968C.cpp
use std::io::{self, BufRead};

const N: usize = 500;
const A: usize = 1_000_000_000;

fn proc() {
    let stdin = io::stdin();
    let reader = stdin.lock();
    let mut lines = reader.lines();
    
    let n: usize = lines.next().unwrap().unwrap().parse().unwrap();
    let mut x = vec![0; n];
    let mut a = vec![0; n];

    for i in 1..n {
        x[i] = lines.next().unwrap().unwrap().parse().unwrap();
    }
    
    a[0] = x[1] + 1;

    for i in 1..n {
        for j in 0..=A {
            a[i] = x[i] + j * a[i - 1];
            if (if i + 1 < n { x[i + 1] } else { 1 }) < a[i] && a[i] != a[i - 1] {
                break;
            }
        }
    }

    for value in &a {
        print!("{} ", value);
    }
    println!();
}

fn main() {
    let stdin = io::stdin();
    let reader = stdin.lock();
    let mut lines = reader.lines();
    
    let t: usize = lines.next().unwrap().unwrap().parse().unwrap();
    for _ in 0..t {
        proc();
    }
}

File: ga/glm4_result_2024_6/result/1968C_glm/16_1968C.rs
use std::io;

fn main() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    let t: i32 = input.trim().parse().unwrap();
    
    for _ in 0..t {
        input.clear();
        io::stdin().read_line(&mut input).unwrap();
        let n: i32 = input.trim().parse().unwrap();
        let mut x = vec![0; n as usize];
        let mut a = vec![0; n as usize];

        for i in 1..n {
            input.clear();
            io::stdin().read_line(&mut input).unwrap();
            x[i as usize] = input.trim().parse().unwrap();
        }

        a[0] = x[1] + 1;

        for i in 1..n {
            for j in 0..=1000000000 {
                a[i as usize] = x[i as usize] + j * a[(i - 1) as usize];
                if (i + 1 < n && x[(i + 1) as usize] < a[i as usize]) || (i + 1 == n && 1 < a[i as usize]) {
                    break;
                }
            }
        }

        for i in 0..n {
            print!("{} ", a[i as usize]);
        }
        println!();
    }
}